src/main/java/graphql/language/AbstractNode.java

```diff
@@ -29,4 +31,31 @@ public void setComments(List<Comment> comments) {
         assertNotNull(comments, "You must provide non null comments");
         this.comments = comments;
     }
+
+    protected boolean isEqualTo(String thisStr, String thatStr) {
+        if (null == thisStr) {
+            if (null != thatStr) {
+                return false;
+            }
+        } else if (!thisStr.equals(thatStr)) {
+            return false;
+        }
+        return true;
+    }
+
+    protected <V> V deepCopy(V nullableObj, Function<V, V> copyFunction) {
```

Comment:
what is the purpose of that? it is only a null check, isn't it?

Reply:
yes but its sugar to make all the "deepCopy callers" look nicer.

eg instead of

    memberVariable == null ? null : memberVariable.deepCopy()

its
   deepCopy(memberVariable, Value:deepCopy)

with casting we can make this smaller - might give that a go